runtime/etc/Makefile.launcher

```diff
@@ -55,7 +55,7 @@ endif
 ifneq ($(CHPL_MAKE_IS_MLI),1)
 	$(CC) $(GEN_CFLAGS) $(COMP_GEN_CFLAGS) \
 		-c -o $(TMPBINNAME)_launcher.o \
-		-DLAUNCHER $(CHPL_LN_INCS) -I. $(LAUNCHER_SRC_NAME)
+		-DLAUNCHER $(CHPL_LN_INCS) -iquote. $(LAUNCHER_SRC_NAME)
```

Comment:
In the compiler change above, we know the compiler is `clang`, so we know it supports `-iquote`. But here in the Makefiles, we have to worry that some C compiler might not support `-iquote`. I am OK with proceeding even so, but please watch for issues as a result.

Reply:
Given that the switch to iquote is only motivated by C++ code and all our launchers are C only, I think the more conservative approach is to just not make this change

Reply:
That's true for this line, but my comment applies to all of the Makefile changes.

Reply:
Looking closer at the other Makefile changes, those Makefiles are only ever used with the C backend or with C source files under our control. So there is no danger of invoking them with a C++ compiler, creating the issue

Reply:
Pedantically, one could have a bogus `stdio.h` file in the current directory. It just wouldn't be as likely as having `array`.

Reply:
Yes I agree, but to handle that case I actually need to expand this PR for `runtime/etc/Makefile.exe` as well.

I am fine with doing that, but what is the benefit to fixing this edge case vs the portability issue of a compile potentially not supporting iquote?

Maybe this is not a big deal, iquote seems to have been around for [gcc since 7.4.0](https://gcc.gnu.org/onlinedocs/gcc-7.4.0/gcc/Directory-Options.html) and for [clang since 11.0.0](https://releases.llvm.org/11.0.0/tools/clang/docs/ClangCommandLineReference.html#cmdoption-clang-iquote-directory)

Reply:
Yes iquote is probably available everywhere that it matters, but we have to watch out for issues with it in testing

Reply:
Ok, I have pushed the additional changes. Once this is merged, I will keep an eye out for portability issues. `icc` comes to mind as a possible pain point, but at least newish versions seem to support it: https://www.intel.com/content/www/us/en/docs/cpp-compiler/developer-guide-reference/2021-8/iquote.html